### PR TITLE
feat(chart-advisor): support define fieldTypes when call advicesForChart

### LINKED ISSUE #549: ChartAdvise enhance to support define fieldtypes info
* **Package Name**: chart-advisor
* **Package Version**: master
* **Platform**:
* **Mini Showcase(like screenshots)**:
* **CodePen Link**:

I want to use advicesForChart function in chart-advisor package,
According to the ChartAdviseParams define, I can provide `data` and `fields` tells which fields to focus
image
Then advicesForChart will create DataFrame use the data and fields
image

 but according to DataFrame, it needs both `columns` and `columnTypes` to make the column correct types.
So I think it should add `columnTypes` in ChartAdviseParams and send it when create DataFrame

### PR DESCRIPTION
**Pull request by zhangxy-zju**
### PR includes

- [x] fixed #549
- [ ] add / modify test cases
- [ ] documents, demos

### CODE REVIEW

File: packages/chart-advisor/src/advisor/index.ts [hunk 1]
```diff
@@ -15,13 +15,21 @@ import type { Advice, AdvicesWithLog, DataRows, GraphData as GData, Data } from
 
 export type AdviseParams = ChartAdviseParams | GraphAdviseParams;
 
+/**
+ * Field Type
+ * @public
+ */
+export declare type TypeSpecifics = 'null' | 'boolean' | 'integer' | 'float' | 'date' | 'string';
```
Comment (pddpd) [inline packages/chart-advisor/src/advisor/index.ts:22]: u could import it from `@antv/data-wizard`'s analyzer

File: packages/chart-advisor/src/advisor/index.ts [hunk 2]
```diff
@@ -147,12 +155,12 @@ export class Advisor {
...
     // otherwise the input data will be mutated
     const copyData = cloneDeep(data);
-    const { fields } = params as ChartAdviseParams;
+    const { fields, fieldTypes } = params as ChartAdviseParams;
     // transform data into DataFrame
     let dataFrame: DataFrame;
     try {
       if (fields) {
-        dataFrame = new DataFrame(copyData, { columns: fields });
+        dataFrame = new DataFrame(copyData, { columns: fields, columnTypes: fieldTypes });
```
Comment (pddpd) [inline packages/chart-advisor/src/advisor/index.ts:163]: Plz note that DW will directly convert the data type, not just modify the info.

File: packages/chart-advisor/src/advisor/index.ts (hunk 1 above)
Comment (zhangxy-zju) [reply packages/chart-advisor/src/advisor/index.ts]: Done

File: packages/chart-advisor/src/advisor/index.ts (hunk 2 above)
Comment (zhangxy-zju) [reply packages/chart-advisor/src/advisor/index.ts:157]: Yes，it is used only in the case that the data is not formated to correct type， for example data is all string from backend
If we don't support this, I need to convert the data type by another new DataFrame which will take wasted effort. Maybe we can add comment to tell user about this

Comment (pddpd) [reply packages/chart-advisor/src/advisor/index.ts:157]: got it, we will also add more information in v3 docs.

File: packages/chart-advisor/src/advisor/index.ts [hunk 3]
```diff
@@ -22,6 +22,8 @@ export type ChartAdviseParams = {
   dataProps?: BasicDataPropertyForAdvice[];
   /** data fields to focus, apply in `data` and `dataProps` */
   fields?: string[];
+  /** specify data types for fields to focus, Please notice that this will modify the data you provided. It will convert the data according to fieldTypes  */
+  fieldTypes?: (analyzer.TypeSpecifics | '')[];
```
Comment (pddpd) [inline packages/chart-advisor/src/advisor/index.ts:26]: Thx for mention, dataset should export types like `Extra['columnTypes']`.
image

**Review by pddpd — APPROVED** [review]
LGTM

**Comment by pddpd** [comment]
@zhangxy-zju  Do you want to modify your git user?

**Comment by GuangMingYouBei** [comment]
A point should be clarified: what is your primary motivation to defined fieldTypes?

-  FiledTypes generated by default may lead to false chart type recommendation, so you want to assign fieldTypes directly.
-  You want to transform the original data. Refer: https://github.com/antvis/AVA/blob/394f49d11e7a84f65c373d13a4c6009e1f3546c1/packages/data-wizard/src/dataset/utils.ts#L73
- Both of the above

**Comment by zhangxy-zju** [comment]
> A point should be clarified: what is your primary motivation to defined fieldTypes? @zhangxy-zju
> 
> * FiledTypes generated by default may lead to false chart type recommendation, so you want to assign fieldTypes directly.
> * You want to transform the original data. Refer: https://github.com/antvis/AVA/blob/394f49d11e7a84f65c373d13a4c6009e1f3546c1/packages/data-wizard/src/dataset/utils.ts#L73-L94
> * Both of the above

Both of them. We have our own logic to decide the field type, one field which looks like all numbers for example id and date should be treat as string to get correct chart. So we want to both use the field type we provided to  transform data,  also including use the field type to advice chart

**Comment by GuangMingYouBei** [comment]
> 

I got it. This PR works for you. 
Meanwhile, we are moving to v3 version, which will be merged to master on 21nd this month as planned. 
We will also support your demand in v3, but parameters could be a little different.

**Pull request merged**